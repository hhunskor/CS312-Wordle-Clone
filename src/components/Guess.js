import PropTypes from "prop-types";
// import { useEffect } from "react";

function Guess({
  input,
  correctWord,
  setGameOver,
  correctLetters,
  setCorrectLetters,
  guessedLetters,
  setGuessedLetters,
  guessedWords,
  setGuessedWords,
}) {
  setGuessedWords(
    [...guessedWords, input].map((word) => {
      return <li key={word}>{word}</li>;
    })
  );

  const correctLettersCopy = correctLetters.map((x) => x);
  const guessedLettersCopy = guessedLetters.map((z) => z);

  for (let index = 0; index < input.length; index++) {
    const letter = input.charAt(index);

    if (correctWord.includes(letter) && !correctLettersCopy.includes(letter)) {
      correctLettersCopy.push(letter);
    }

    if (!guessedLettersCopy.includes(letter)) {
      guessedLettersCopy.push(letter);
    }
  }

  setCorrectLetters(correctLettersCopy);
  setGuessedLetters(guessedLettersCopy);

  //console.log(correctLetters);
  //console.log(guessedLetters);

  //check if word is correct
  if (input === correctWord) {
    setGameOver(true);
  }

  console.log(guessedWords);
  return <div>{guessedWords}</div>;
}

Guess.propTypes = {
  input: PropTypes.string.isRequired,
  correctWord: PropTypes.string.isRequired,
  setGameOver: PropTypes.func.isRequired,
  correctLetters: PropTypes.arrayOf(PropTypes.string).isRequired,
  setCorrectLetters: PropTypes.func.isRequired,
  guessedLetters: PropTypes.arrayOf(PropTypes.string).isRequired,
  setGuessedLetters: PropTypes.func.isRequired,
  guessedWords: PropTypes.arrayOf(PropTypes.string).isRequired,
  setGuessedWords: PropTypes.func.isRequired,
};

export default Guess;
/* eslint-disable */
