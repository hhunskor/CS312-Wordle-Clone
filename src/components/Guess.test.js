import { render, screen } from "@testing-library/react";

import Guess from "../components/Guess";

const tiles = [
  {
    tile: 0,
    letter: "",
    color: "black",
  },
  {
    tile: 1,
    letter: "",
    color: "black",
  },
  {
    tile: 2,
    letter: "",
    color: "black",
  },
  {
    tile: 3,
    letter: "",
    color: "black",
  },
  {
    tile: 4,
    letter: "",
    color: "black",
  },
  {
    tile: 5,
    letter: "",
    color: "black",
  },
  {
    tile: 6,
    letter: "",
    color: "black",
  },
  {
    tile: 7,
    letter: "",
    color: "black",
  },
  {
    tile: 8,
    letter: "",
    color: "black",
  },
  {
    tile: 9,
    letter: "",
    color: "black",
  },
  {
    tile: 10,
    letter: "",
    color: "black",
  },
  {
    tile: 11,
    letter: "",
    color: "black",
  },
  {
    tile: 12,
    letter: "",
    color: "black",
  },
  {
    tile: 13,
    letter: "",
    color: "black",
  },
  {
    tile: 14,
    letter: "",
    color: "black",
  },
  {
    tile: 15,
    letter: "",
    color: "black",
  },
  {
    tile: 16,
    letter: "",
    color: "black",
  },
  {
    tile: 17,
    letter: "",
    color: "black",
  },
  {
    tile: 18,
    letter: "",
    color: "black",
  },
  {
    tile: 19,
    letter: "",
    color: "black",
  },
  {
    tile: 20,
    letter: "",
    color: "black",
  },
  {
    tile: 21,
    letter: "",
    color: "black",
  },
  {
    tile: 22,
    letter: "",
    color: "black",
  },
  {
    tile: 23,
    letter: "",
    color: "black",
  },
  {
    tile: 24,
    letter: "",
    color: "black",
  },
  {
    tile: 25,
    letter: "",
    color: "black",
  },
  {
    tile: 26,
    letter: "",
    color: "black",
  },
  {
    tile: 27,
    letter: "",
    color: "black",
  },
  {
    tile: 28,
    letter: "",
    color: "black",
  },
  {
    tile: 29,
    letter: "",
    color: "black",
  },
];

describe("Guesses visibility", () => {
  test("Smoke test", () => {
    render(<Guess tiles={tiles} />);
  });
  test("Snapshot test", () => {
    const { container } = render(<Guess tiles={tiles} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

const sampleGuesses = ["lines", "bagel", "match", "chart"];

describe("Guesses render", () => {
  test("Guesses populate based on tile elements", () => {
    sampleGuesses.forEach((sampleGuess) => {
      for (let i = 0; i < sampleGuess.length; i++) {
        const guessedLetter = sampleGuess.charAt(i).toUpperCase();
        tiles[i] = {
          tile: i,
          letter: guessedLetter,
          color: "black",
        };
      }
      render(<Guess tiles={tiles} />);
      const uppercaseLetter = sampleGuess.charAt(0).toUpperCase();
      // console.log(uppercaseLetter);
      expect(screen.getByText({ uppercaseLetter })).toBeVisible();
    });
  });
});
