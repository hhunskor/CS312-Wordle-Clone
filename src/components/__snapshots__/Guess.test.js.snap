// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Guesses visibility Snapshot test 1`] = `
<div
  id="grid"
>
  <div
    class="row"
  >
    <div
      class="initBox"
    >
       
       
    </div>
    <div
      class="initBox"
    >
       
       
    </div>
    <div
      class="initBox"
    >
       
       
    </div>
    <div
      class="initBox"
    >
       
       
    </div>
    <div
      class="initBox"
    >
       
       
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="initBox"
    >
       
       
    </div>
    <div
      class="initBox"
    >
       
       
    </div>
    <div
      class="initBox"
    >
       
       
    </div>
    <div
      class="initBox"
    >
       
       
    </div>
    <div
      class="initBox"
    >
       
       
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="initBox"
    >
       
       
    </div>
    <div
      class="initBox"
    >
       
       
    </div>
    <div
      class="initBox"
    >
       
       
    </div>
    <div
      class="initBox"
    >
       
       
    </div>
    <div
      class="initBox"
    >
       
       
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="initBox"
    >
       
       
    </div>
    <div
      class="initBox"
    >
       
       
    </div>
    <div
      class="initBox"
    >
       
       
    </div>
    <div
      class="initBox"
    >
       
       
    </div>
    <div
      class="initBox"
    >
       
       
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="initBox"
    >
       
       
    </div>
    <div
      class="initBox"
    >
       
       
    </div>
    <div
      class="initBox"
    >
       
       
    </div>
    <div
      class="initBox"
    >
       
       
    </div>
    <div
      class="initBox"
    >
       
       
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="initBox"
    >
       
       
    </div>
    <div
      class="initBox"
    >
       
       
    </div>
    <div
      class="initBox"
    >
       
       
    </div>
    <div
      class="initBox"
    >
       
       
    </div>
    <div
      class="initBox"
    >
       
       
    </div>
  </div>
</div>
`;
